indico/modules/events/papers/operations.py

```diff
@@ -57,3 +61,21 @@ def update_competences(user_competences, competences):
     event.log(EventLogRealm.management, EventLogKind.positive, 'Papers',
               "Updated competences for user {}".format(user_competences.user.full_name), session.user,
               data={'Competences': ', '.join(competences)})
+
+
+def paper_change_state(contribution, contrib_data, judgment, judge, send_notifications=False):
+    log_data = {'New state': orig_string(judgment.title)}
+    if judgment == PaperAction.accept:
+        contribution.paper_last_revision.state = PaperRevisionState.accepted
+    elif judgment == PaperAction.reject:
+        contribution.paper_last_revision.state = PaperRevisionState.rejected
+    elif judgment == PaperAction.to_be_corrected:
+        contribution.paper_last_revision.state = PaperRevisionState.to_be_corrected
+    PaperReviewComment(paper_revision=contribution.paper_last_revision, user=judge,
```

Comment:
So we do want to save the judgment comment  as a normal comment in the end and not in a separate field on the revision?

Anyway, while this works (due to being added to the SA session since you link it to he prevision) I think it's slighly ugly to create an instance without using it anywhere. I'd rather do it like this:

    if contrib_data['judgment_comment']:
        comment = PaperReviewComment(user=judge, text=contrib_data['judgment_comment'])
        contribution.paper_last_revision.comments.append(comment)